### DISCUSSION TITLE
Caret notation can't be used (at least in /tp command)

### DISCUSSION THREAD

**Discussion by MetaZ9** [EssentialsX/Essentials#4175] [Ideas and Feature Suggestions] [answered]
### Type of bug

Other unexpected behaviour

### `/ess version` output

[Server thread/INFO]: Version Server : 1.16.5-R0.1-SNAPSHOT 2991-Bukkit-f3f3094 (MC: 1.16.5)
[Server thread/INFO]: Version EssentialsX : 2.18.2.0
[Server thread/INFO]: Vault is not installed. Chat and permissions may not work.
[Server thread/INFO]: You are using an unsupported server version!

### Server startup log

https://paste.gg/p/anonymous/70b34a933abb4ce79dac6f78724c88df

### EssentialsX config files

https://paste.gg/p/anonymous/d2a388e4d5fc48ab833aa71ee9c51b39

### Bug description

When trying to use the caret notation for local coordinates in /tp, I get the result : 'Error : For input string "^"'

### Steps to reproduce

Run `/tp ^ ^ ^5`

### Expected behaviour

I should be teleported 5 blocks forward in the direction I'm looking at

### Actual behaviour

I get 'Error : For input string "^"' and nothing happens

**Comment by pop4959** [comment]
This isn't really a bug, as Essentials currently doesn't support (nor has it ever) carets in any teleportation commands. The teleport command in vanilla supports this however, which you can use by running `/minecraft:tp` instead of Essentials' `/tp`.

**Comment by triagonal** [comment]
If I'm not mistaken, you can use `~` in place of `^` in EssentialsX teleport commands. It should have the same (or very similar) effect.

**Comment by JRoy** [comment, marked as answer]
> If I'm not mistaken, you can use `~` in place of `^` in EssentialsX teleport commands. It should have the same (or very similar) effect.

They're not quite the same, [caret notion](https://minecraft.fandom.com/wiki/Commands#Local_coordinates:_Caret_notation) represents local coordinates (based on where the player's camera is facing), where the tilde notation represents relative world coordinates.

We could *probably* add support for caret notion, I'll move this to a feature request for now

**Reply by MetaZ9** [reply]
I'll be grateful for that. In fact, we could always use the vanilla Minecraft commands by prefixing them or disabling the relevent Essentials commands if we want to use caret notation; but I find this rather troublesome and not really efficient or relevant in this case
Thanks for the answer
